### PR TITLE
feat: material mapper in core (MM4)

### PR DESCRIPTION
**Pull request by asalzburger**
This is the first PR in a series that divides the Material Mapping into logical, unit testable modules:

1) Finding intersections with surfaces and associations to volumes (MM2)
2) Assigning material interactions to those intersections (MM1)
3) Mapping those onto dedicated Surface / Volume Material Mappers (MM3)
4) Steer that by a chained algorithm (MM4, this PR)

This PR:

Introduces a MaterialMapper which just needs I/O connection in the relevant framework/experiment context implementation.

All cases are showcased and tested in a set of UnitTests.

### CODE REVIEW

**Review by Corentin-Allaire — APPROVED** [review]
Looks good ! Is the plan to add the volume part later ?

**Comment by asalzburger** [comment]
> Looks good ! Is the plan to add the volume part later ?

Yes, I will finish the surface part first, and try to make it fully reproduce what we had before.

File: Core/src/Material/MaterialMapper.cpp [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+//
+// Copyright (C) 2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "Acts/Material/MaterialMapper.hpp"
+
+#include "Acts/Utilities/Enumerate.hpp"
```
[the remark this answers is not in the document]
Comment (asalzburger) [reply Core/src/Material/MaterialMapper.cpp:11]: will remove in next PR.

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results [21cbbf749f9171a7616d9c9210506be7a8d83f05]
### :white_check_mark: All tests successful

|status|job|report|
|------|---|------|

**Pull request merged**